src/github-installer.ts

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as http from 'http';
+import * as https from 'https';
+import * as os from 'os';
+import { URL } from 'url';
+import * as vscode from 'vscode';
+
+import { opaOutputChannel } from './extension';
+const releaseDownloader = require('@fohlen/github-release-downloader');
+
+let installDeclined: { [repo: string]: boolean } = {};
+
+export function promptForInstall(
+    binaryConfigKey: string,
+    repo: string,
+    message: string,
+    determineBinaryURLFromRelease: (release: any) => string,
+    determineExecutableName: () => string,
+) {
+    if (installDeclined[binaryConfigKey]) {
+        return;
+    }
+
+    vscode.window.showInformationMessage(message, 'Install')
+        .then((selection) => {
+            if (selection === 'Install') {
+                install(
+                    binaryConfigKey,
+                    repo,
+                    determineBinaryURLFromRelease,
+                    determineExecutableName,
+                );
+            } else {
+                installDeclined[binaryConfigKey] = true;
+            }
+        });
+}
+
+async function install(binaryConfigKey: string, repo: string, determineBinaryURLFromRelease: (release: any) => string, determineExecutableName: () => string) {
```

Comment:
Very imperative, but also very easy to follow!

Reply:
it does the job, mostly just copy pasted from the old OPA installer, with some parts made more generic.